pyfolio/risk.py

```diff
@@ -0,0 +1,486 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import matplotlib.pyplot as plt
+from collections import OrderedDict
+
+SECTORS = OrderedDict([
+    (101, '101 Basic Materials'),
+    (102, '102 Consumer Cyclical'),
+    (103, '103 Financial Services'),
+    (104, '104 Real Estate'),
+    (205, '205 Consumer Defensive'),
+    (206, '206 Healthcare'),
+    (207, '207 Utilities'),
+    (308, '308 Communication Services'),
+    (309, '309 Energy'),
+    (310, '310 Industrials'),
+    (311, '311 Technology')
+])
+
+CAP_CUTOFFS = [50000000, 300000000, 2000000000, 10000000000, 200000000000]
+CAP_NAMES = ['Micro', 'Small', 'Mid', 'Large', 'Mega']
+
+
+def compute_style_factor_exposures(positions, risk_factor):
+    '''
+    Returns style factor exposure of an algorithm's positions
+
+    Parameters
+    ----------
+    positions : pd.DataFrame
+        Daily equity positions of algorithm, in dollars.
+        - See full explanation in create_risk_tear_sheet
+
+    risk_factor : pd.DataFrame
+        Daily risk factor per asset.
+        - DataFrame with dates as index and equities as columns
+        - Example:
+                     Equity(24   Equity(62
+                       [AAPL])      [ABT])
+        2017-04-03	  -0.51284     1.39173
+        2017-04-04	  -0.73381     0.98149
+        2017-04-05	  -0.90132	   1.13981
+    '''
+
+    positions_wo_cash = positions.drop('cash', axis=1)
+    gross_exposure = positions_wo_cash.abs().sum(axis=1)
+
+    sfe = positions_wo_cash.multiply(risk_factor, level=1) \
+        .divide(gross_exposure, axis='index')
+    tot_sfe = sfe.sum(axis=1, skipna=True)
+
+    return tot_sfe
+
+
+def plot_style_factor_exposures(tot_sfe, factor_name, ax=None):
+    '''
+    Plots DataFrame output of compute_style_factor_exposures as a line graph
+
+    Parameters
+    ----------
+    tot_sfe : pd.Series
+        Daily style factor exposures (output of compute_style_factor_exposures)
+        - Time series with decimal style factor exposures
+        - Example:
+            2017-04-24    0.037820
+            2017-04-25    0.016413
+            2017-04-26   -0.021472
+            2017-04-27   -0.024859
+
+    factor_name : string
+        Name of style factor, for use in graph title
+    '''
+
+    if ax is None:
+        ax = plt.gca()
+
+    ax.plot(tot_sfe.index, tot_sfe, label=factor_name)
+    avg = tot_sfe.mean()
+    ax.axhline(avg, linestyle='-.', label='Mean = {:.3}'.format(avg))
+    ax.axhline(0, color='k', linestyle='-')
+    _, _, y1, y2 = plt.axis()
+    a = max(abs(y1), abs(y2))
+    ax.set_ylim(-a, a)
```

Comment:
Did not know that `ax` had that function! Replaced all instances of `ax.set_foo(...); ax.set_bar(...)` with `ax.set(foo=..., bar=...)